LibAddonMenu-2.0/controls/dropdown.lua

```diff
@@ -1,439 +1,463 @@
+    local allItemsHeight = (dropdown:GetEntryTemplateHeightWithSpacing() * numEntries) - entrySpacing + (PADDING_Y * 2) + ROUNDING_MARGIN
+    dropdown:SetHeight(allItemsHeight)
+    ZO_ScrollList_Commit(dropdown.m_scroll)
+
+    return visibleRows, min, max
+end
+
+function LAMCreateControl.dropdown(parent, dropdownData, controlName)
+    local control = LAM.util.CreateLabelAndContainerControl(parent, dropdownData, controlName)
+    control.choices = {}
+
+    local countControl = parent
+    local name = parent:GetName()
+    if not name or #name == 0 then
+        countControl = LAMCreateControl
+        name = "LAM"
+    end
+    local comboboxCount = (countControl.comboboxCount or 0) + 1
+    countControl.comboboxCount = comboboxCount
+    control.combobox = wm:CreateControlFromVirtual(zo_strjoin(nil, name, "Combobox", comboboxCount), control.container, "ZO_ComboBox")
+
+    local combobox = control.combobox
+    combobox:SetAnchor(TOPLEFT)
+    combobox:SetDimensions(control.container:GetDimensions())
+    combobox:SetHandler("OnMouseEnter", function() ZO_Options_OnMouseEnter(control) end)
+    combobox:SetHandler("OnMouseExit", function() ZO_Options_OnMouseExit(control) end)
+    control.dropdown = ZO_ComboBox_ObjectFromContainer(combobox)
+    local dropdown = control.dropdown
+    dropdown:SetSortsItems(false) -- need to sort ourselves in order to be able to sort by value
+    dropdown.m_dropdown:SetParent(combobox:GetOwningWindow()) -- TODO remove workaround once the problem is fixed in the game
+
+
+    local isMultiSelectionEnabled = GetDefaultValue(dropdownData.multiSelect)
+    control.isMultiSelectionEnabled = isMultiSelectionEnabled
+
+    dropdown.CalculateContentWidth = function()
+        local dataType = ZO_ScrollList_GetDataTypeTable(dropdown.m_scroll, 1)
+
+        local dummy = dataType.pool:AcquireObject()
+        dataType.setupCallback(dummy, {
+            m_owner = dropdown,
+            name = "Dummy"
+        }, dropdown)
+
+        local maxWidth = 0
+        local label = dummy.m_label
+        local entries = dropdown.m_sortedItems
+        local numItems = #entries
+        for index = 1, numItems do
+            label:SetText(entries[index].name)
+            local width = label:GetTextWidth()
+            if (width > maxWidth) then
+                maxWidth = width
+            end
+        end
+
+        dataType.pool:ReleaseObject(dummy.key)
+        return maxWidth
+    end
+
+    dropdown.AdjustDimensions = function()
+        local numItems = #dropdown.m_sortedItems
+        local dropdownObject = dropdown.m_dropdown
+        local scroll = dropdownObject:GetNamedChild("Scroll")
+        local scrollContent = scroll:GetNamedChild("Contents")
+        local anchorOffset = 0
+
+        local contentWidth = dropdown.CalculateContentWidth() + CONTENT_PADDING
+        local visibleRows, min, max = SetDropdownHeight(combobox, dropdown, dropdownData)
+
+        if(numItems > visibleRows) then

```

Comment:
We don't do parentheses in Lua if statements unless it's required for the logic to work

Reply:
Well this was copied from original LAM 2.0 r35 code :-)

if(numItems > self.visibleRows) then